cpp/daal/src/algorithms/dbscan/dbscan_partial_result_types.cpp

```diff
@@ -152,26 +152,35 @@ Status DistributedPartialResultStep4::check(const daal::algorithms::Input * inpu
     DataCollectionPtr dcPartitionedPartialOrders = get(partitionedPartialOrders);
 
     DAAL_CHECK_EX(dcPartitionedData.get(), ErrorNullPartialResult, ArgumentName, partitionedDataStr());
-    DAAL_CHECK_EX(dcPartitionedData->size() == nBlocks, ErrorIncorrectDataCollectionSize, ArgumentName, partitionedDataStr());
+    if (NumericTable::cast((*dcPartitionedData)[0])->getNumberOfRows() != 0)
+    {
+        DAAL_CHECK_EX(dcPartitionedData->size() == nBlocks, ErrorIncorrectDataCollectionSize, ArgumentName, partitionedDataStr());
+    }
 
     DAAL_CHECK_EX(dcPartitionedPartialOrders.get(), ErrorNullPartialResult, ArgumentName, partitionedPartialOrdersStr());
-    DAAL_CHECK_EX(dcPartitionedPartialOrders->size() == nBlocks, ErrorIncorrectDataCollectionSize, ArgumentName, partitionedPartialOrdersStr());
+    if (NumericTable::cast((*dcPartitionedPartialOrders)[0])->getNumberOfRows() != 0)
+    {
+        DAAL_CHECK_EX(dcPartitionedPartialOrders->size() == nBlocks, ErrorIncorrectDataCollectionSize, ArgumentName, partitionedPartialOrdersStr());
+    }
 
     const int unexpectedLayouts = (int)packed_mask;
 
-    for (size_t i = 0; i < nBlocks; i++)
+    if (NumericTable::cast((*dcPartitionedData)[0])->getNumberOfRows() != 0)
```

Comment:
it is reasonable to put this check inside loop and just skip tables with 0 rows.